### ISSUE TITLE
XSS using data-url when doing SSR

### ISSUE THREAD

**Issue by valterkraemer** [sveltejs/kit#2530]
### Describe the bug

It is possible to inject a script tag in `data-url` in the `svelte-data` script tag.

E.g. if we have the search term in the query params https://www.google.com/search?q=Svelte we can create a malicious term that inject a script tag while SSR.

Happens on Firefox, Brave and Chrome, but Safari seems to protect against my POC, however it should be possible to circumvent it in other scenarios.

##### term.js
```js
export const term = '/asdf?search="></script><script>alert(1)</script><script type="application/json';
```

##### index.svelte
```html
<script context="module">
  import { term } from './term';

  export async function load({ fetch }) {
    const res = await fetch(term);

    return {
      props: {
        results: await res.json()
      }
    };
  }
</script>

<h1>Supersearch</h1>

```

#### Deployed POC using query params

POC Deployed: https://debug-app.vkrae.workers.dev/?search=%22%3E%3C/script%3E%3Cscript%3Ealert(1)%3C/script%3E%3Cscript%20type=%22application/json

POC Code: https://github.com/valterkraemer/debug-app/commit/1cc04c89d033bd31c0fb3ccb88ef8bb039128507

### System Info

```shell
System:
  OS: macOS 11.6
  CPU: (8) arm64 Apple M1
  Memory: 144.72 MB / 8.00 GB
  Shell: 5.8 - /bin/zsh
Binaries:
  Node: 14.17.6 - ~/.nvm/versions/node/v14.17.6/bin/node
  npm: 6.14.15 - ~/.nvm/versions/node/v14.17.6/bin/npm
Browsers:
  Brave Browser: 93.1.29.79
  Chrome: 94.0.4606.61
  Firefox: 92.0.1
  Safari: 14.1.2
npmPackages:
  @sveltejs/adapter-cloudflare-workers: ^1.0.0-next.21 => 1.0.0-next.23 
  @sveltejs/kit: next => 1.0.0-next.178 
  svelte: ^3.42.6 => 3.43.1
```

### Severity

serious, but I can work around it

**Comment by Conduitry** [comment]
This is a good catch, thanks!

The `data-url` attribute rendered at https://github.com/sveltejs/kit/blob/d8efaa2faa9c7a93c9b7e7cf234bc485de330d28/packages/kit/src/runtime/server/page/render.js#L171 should presumably be HTML-entity-escaped.

**Comment by Conduitry** [comment]
While we're at it, we should probably also take a look at where the JSON inside the attribute is escaped - https://github.com/sveltejs/kit/blob/490e3d9c150eff18d6c5f5050cf47b2baf37b4ad/packages/kit/src/runtime/server/page/load_node.js#L239 - There appear to be some steps taken to make this more safe than just `JSON.stringify()`ing the whole thing, but it's always good to check again.

**Comment by Conduitry** [comment]
Thank you for the report! This should be fixed in `1.0.0-next.179`.

**Comment by valterkraemer** [comment]
Great! That was quick! ☺️
